src/main/java/core/models/SocketPayload.java

```diff
@@ -9,8 +13,9 @@
     private final InetAddress address;
     private final Socket socket;
 
-    public SocketPayload(String message, int port, InetAddress address, Socket socket) {
-        this.payloadMessage = new PayloadMessage(message);
+    public SocketPayload(String message, int port, InetAddress address, Socket socket) throws IOException {
```

Comment:
Considered this with folks and we agree that this class is useless, we have a port and adress in `socket` object, so we can easly move property socket to message class